src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/IteratorAndAsyncCaptureWalker.cs

```diff
@@ -194,7 +194,12 @@ private void CaptureVariable(Symbol variable, SyntaxNode syntax)
             var type = (variable.Kind == SymbolKind.Local) ? ((LocalSymbol)variable).Type : ((ParameterSymbol)variable).Type;
             if (type.IsRestrictedType())
             {
-                (_lazyDisallowedCaptures ??= new MultiDictionary<Symbol, SyntaxNode>()).Add(variable, syntax);
+                // Interpolated string builder variables are limited in lifetime, and will not be used in a context that
+                // actually needs to be saved to the closure.
+                if (variable is not SynthesizedLocal { SynthesizedKind: SynthesizedLocalKind.InterpolatedStringBuilder })
```

Comment:
Another case where we're treating the synthesized locals differently depending on the underlying kind :/